### ISSUE TITLE
Unable to use AzureSDConfig with ManagedIdentity

### ISSUE THREAD

**Issue by micborens** [prometheus-operator/prometheus-operator#6256]
### Is there an existing issue for this?
No
- [X] I have searched the existing issues

## Description

Hello, we started to setup prometheus and a scrapeconfig with azuresdcofnig but we are getting some issues.

Optional fields when using AzureSDConfig with `AuthenticationMethod: ManagedIdentity` are marked as required
We looked at the code and we found a typo in the check https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/resource_selector.go#L915

`ManagedIdentify` instead of `ManagedIdentity`

### Prometheus Operator Version

```shell
Name:                   prometheus-operator-kube-p-operator
Namespace:              prometheus-operator
CreationTimestamp:      Mon, 22 Jan 2024 13:37:09 +0100
Labels:                 app=kube-prometheus-stack-operator
                        app.kubernetes.io/instance=prometheus-operator
                        app.kubernetes.io/managed-by=Helm
                        app.kubernetes.io/part-of=kube-prometheus-stack
                        app.kubernetes.io/version=55.8.1
                        chart=kube-prometheus-stack-55.8.1
                        heritage=Helm
                        prometheus.io/operator=true
                        release=prometheus-operator
Annotations:            deployment.kubernetes.io/revision: 1
                        meta.helm.sh/release-name: prometheus-operator
                        meta.helm.sh/release-namespace: prometheus-operator
Selector:               app=kube-prometheus-stack-operator,release=prometheus-operator
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           app=kube-prometheus-stack-operator
                    app.kubernetes.io/instance=prometheus-operator
                    app.kubernetes.io/managed-by=Helm
                    app.kubernetes.io/part-of=kube-prometheus-stack
                    app.kubernetes.io/version=55.8.1
                    chart=kube-prometheus-stack-55.8.1
                    heritage=Helm
                    prometheus.io/operator=true
                    release=prometheus-operator
  Service Account:  prometheus-operator-kube-p-operator
  Containers:
   kube-prometheus-stack:
    Image:      harbor-replica.eastus2.csq.io/quay/prometheus-operator/prometheus-operator:v0.70.0
    Port:       8080/TCP
    Host Port:  0/TCP
    Args:
      --kubelet-service=kube-system/prometheus-operator-kube-p-kubelet
      --log-format=json
      --log-level=info
      --localhost=127.0.0.1
      --prometheus-config-reloader=harbor-replica.eastus2.csq.io/quay/prometheus-operator/prometheus-config-reloader:v0.70.0
      --config-reloader-cpu-request=200m
      --config-reloader-cpu-limit=0
      --config-reloader-memory-request=60Mi
      --config-reloader-memory-limit=60Mi
      --thanos-default-base-image=quay.io/thanos/thanos:v0.33.0
      --secret-field-selector=type!=kubernetes.io/dockercfg,type!=kubernetes.io/service-account-token,type!=helm.sh/release.v1
    Limits:
      memory:  512Mi
    Requests:
      cpu:     200m
      memory:  512Mi
    Environment:
      GOGC:             30
    Mounts:             <none>
  Volumes:              <none>
  Priority Class Name:  scaling
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      True    MinimumReplicasAvailable
OldReplicaSets:  <none>
NewReplicaSet:   prometheus-operator-kube-p-operator-5dc68669db (1/1 replicas created)
Events:          <none>
```

### Kubernetes Version

```yaml
clientVersion:
  buildDate: "2023-08-24T00:50:44Z"
  compiler: gc
  gitCommit: 395f0a2fdc940aeb9ab88849e8fa4321decbf6e1
  gitTreeState: clean
  gitVersion: v1.26.8
  goVersion: go1.20.7
  major: "1"
  minor: "26"
  platform: darwin/arm64
kustomizeVersion: v4.5.7
serverVersion:
  buildDate: "2023-10-23T08:13:55Z"
  compiler: gc
  gitCommit: a8a1abc25cad87333840cd7d54be2efaf31a3177
  gitTreeState: clean
  gitVersion: v1.28.3
  goVersion: go1.20.10
  major: "1"
  minor: "28"
  platform: linux/amd64

WARNING: version difference between client (1.26) and server (1.28) exceeds the supported minor version skew of +/-1
```

### Kubernetes Cluster Type

AKS

### How did you deploy Prometheus-Operator?

helm chart:prometheus-community/kube-prometheus-stack

### Manifests

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: ScrapeConfig
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"monitoring.coreos.com/v1alpha1","kind":"ScrapeConfig","metadata":{"annotations":{},"labels":{"prometheus.io/operator":"true"},"name":"test-promop-scrapeconfig","namespace":"prometheus-operator"},"spec":{"azureSDConfigs":[{"authenticationMethod":"ManagedIdentity","clientID":"a","clientSecret":{"key":"prometheus.yaml.gz","name":"prometheus-prometheus-operator-kube-p","optional":true},"environment":"AzurePublicCloud","port":9100,"refreshInterval":"60s","subscriptionID":"xxxxxx","tenantID":"xxxxxx"}],"metricsPath":"/metrics"}}
  creationTimestamp: "2024-01-24T09:53:58Z"
  generation: 17
  labels:
    prometheus.io/operator: "true"
  name: test-promop-scrapeconfig
  namespace: prometheus-operator
  resourceVersion: "243457188"
  uid: 3d55334d-9293-47c9-b6fe-6d8531459042
spec:
  azureSDConfigs:
    - authenticationMethod: ManagedIdentity
      clientID: a
      clientSecret:
        key: prometheus.yaml.gz
        name: prometheus-prometheus-operator-kube-p
        optional: true
      environment: AzurePublicCloud
      port: 9100
      refreshInterval: 5m
      subscriptionID: xxxxxx
      tenantID: xxxxx
  metricsPath: /metrics
```

### prometheus-operator log output

```shell
prometheus-prometheus-operator-kube-p-0 prometheus {"caller":"refresh.go:80","component":"discovery manager scrape","config":"scrapeConfig/prometheus-operator/test-promop-scrapeconfig","discovery":"azure","err":"could not get virtual machines: could not list virtual machines: ManagedIdentityCredential authentication failed\nGET http://169.254.169.254/metadata/identity/oauth2/token\n--------------------------------------------------------------------------------\nRESPONSE 400 Bad Request\n--------------------------------------------------------------------------------\n{\n  \"error\": \"invalid_request\",\n  \"error_description\": \"Identity not found\"\n}\n--------------------------------------------------------------------------------\nTo troubleshoot, visit https://aka.ms/azsdk/go/identity/troubleshoot#managed-id","level":"error","msg":"Unable to refresh target groups","ts":"2024-01-24T11:09:20.497Z"}
```

### Anything else?

PR to fix https://github.com/prometheus-operator/prometheus-operator/pull/6259

**Comment by slashpai** [comment]
Thanks for reporting the issue and fix :)

**Comment by slashpai** [comment]
`AuthenticationMethod` is optional field https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go#L496, Prometheus by default uses `OAuth`

Are you able to discover the instances after the fix tested locally?

**Comment by micborens** [comment]
> `AuthenticationMethod` is optional field https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go#L496, Prometheus by default uses `OAuth`
> 
> Are you able to discover the instances after the fix tested locally?

No sorry we didn't have the time to build and test the crd locally. But following the fix, I think it's a waste of time to test it locally.
But after this if we find other issues it will be a pleasure to find and test and fix them :)

**Comment by simonpasquier** [comment]
closed by #6259
